modules/scalacheck/shared/src/main/scala/weaver/scalacheck/Checkers.scala

```diff
@@ -106,22 +106,18 @@ trait Checkers {
 
     private def forall_[A: Show](gen: Gen[A], f: A => F[Expectations])(
         implicit loc: SourceLocation): F[Expectations] = {
-      paramStream
-        .parEvalMap(config.perPropertyParallelism) {
-          testOneTupled(gen, f)
-        }
-        .mapAccumulate(Status.start[A]) { case (oldStatus, testResult) =>
-          val newStatus = testResult match {
+      val (params, initialSeed) = startParams
+      seedStream(initialSeed)
+        .parEvalMap(config.perPropertyParallelism)(testOne(gen, f)(params, _))
+        .scan(Status.start[A]) { case (oldStatus, testResult) =>
+          testResult match {
             case TestResult.Success => oldStatus.addSuccess
             case TestResult.Discard => oldStatus.addDiscard
-            case TestResult.Failure(input, seed, exp) =>
-              oldStatus.addFailure(input, seed, exp)
+            case TestResult.Failure(input, exp) =>
+              oldStatus.addFailure(input, initialSeed, exp)
           }
-          (newStatus, newStatus)
         }
-        .map(_._1)
         .takeWhile(_.shouldContinue(config), takeFailure = true)
-        .takeRight(1) // getting the first error (which finishes the stream)
         .compile
         .last
         .map { (x: Option[Status[A]]) =>
```

Comment:
This stream will always contain at least one value due to the properties of `scan`, so the `None` case will never be hit. 

We could use `lastOrError` here instead. What's your preference?

Reply:
`lastOrError` is good 👍